### PR TITLE
[EncoderDecoder] Improve docs

### PR DESCRIPTION
**Pull request by NielsRogge**
# What does this PR do?

As a follow-up of #17815, this PR improves the docs of `VisionEncoderDecoderModel` and `SpeechEncoderDecoderModel`.

It also fixes some typos in the docs of `EncoderDecoderModel`.

### CODE REVIEW

File: docs/source/en/model_doc/encoder-decoder.mdx [hunk 1]
```diff
@@ -55,14 +55,32 @@ To do so, the `EncoderDecoderModel` class provides a [`EncoderDecoderModel.from_
 >>> model = EncoderDecoderModel.from_encoder_decoder_pretrained("bert-base-uncased", "bert-base-uncased")
 ```
 
-## Loading an existing [`EncoderDecoderModel`] checkpoint.
+## Loading an existing `EncoderDecoderModel` checkpoint and perform inference.
 
-To load fine-tuned checkpoints of the `EncoderDecoderModel` class, ['EncoderDecoderModel`] provides the `from_pretrained(...)` method just like any other model architecture in Transformers.
+To load fine-tuned checkpoints of the `EncoderDecoderModel` class, [`EncoderDecoderModel`] provides the `from_pretrained(...)` method just like any other model architecture in Transformers.
```
Comment (sgugger) [inline docs/source/en/model_doc/encoder-decoder.mdx:60]:
```suggestion
To load fine-tuned checkpoints of the `EncoderDecoderModel` class, [`EncoderDecoderModel`] provides the `from_pretrained()` method just like any other model architecture in Transformers.
```

File: docs/source/en/model_doc/speech-encoder-decoder.mdx [hunk 2]
```diff
@@ -12,17 +12,103 @@ specific language governing permissions and limitations under the License.
...
+>>> from transformers import SpeechEncoderDecoderModel
+
+>>> model = SpeechEncoderDecoderModel.from_encoder_decoder_pretrained(
+...     "facebook/hubert-large-ls960-ft", "bert-base-uncased"
+... )
+```
+
+## Loading an existing `SpeechEncoderDecoderModel` checkpoint and perform inference.
+
+To load fine-tuned checkpoints of the `SpeechEncoderDecoderModel` class, [`SpeechEncoderDecoderModel`] provides the `from_pretrained(...)` method just like any other model architecture in Transformers.
```
Comment (sgugger) [inline docs/source/en/model_doc/speech-encoder-decoder.mdx:57]:
```suggestion
To load fine-tuned checkpoints of the `SpeechEncoderDecoderModel` class, [`SpeechEncoderDecoderModel`] provides the `from_pretrained()` method just like any other model architecture in Transformers.
```

File: docs/source/en/model_doc/vision-encoder-decoder.mdx [hunk 3]
```diff
@@ -12,16 +12,136 @@ specific language governing permissions and limitations under the License.
...
+>>> from transformers import VisionEncoderDecoderModel
+
+>>> model = VisionEncoderDecoderModel.from_encoder_decoder_pretrained(
+...     "microsoft/swin-base-patch4-window7-224-in22k", "bert-base-uncased"
+... )
+```
+
+## Loading an existing `VisionEncoderDecoderModel` checkpoint and perform inference.
+
+To load fine-tuned checkpoints of the `VisionEncoderDecoderModel` class, [`VisionEncoderDecoderModel`] provides the `from_pretrained(...)` method just like any other model architecture in Transformers.
```
Comment (sgugger) [inline docs/source/en/model_doc/vision-encoder-decoder.mdx:63]:
```suggestion
To load fine-tuned checkpoints of the `VisionEncoderDecoderModel` class, [`VisionEncoderDecoderModel`] provides the `from_pretrained()` method just like any other model architecture in Transformers.
```

**Review by sgugger — APPROVED** [review]
Thanks for expanding the doc!

File: docs/source/en/model_doc/speech-encoder-decoder.mdx [hunk 4]
```diff
@@ -12,17 +12,103 @@ specific language governing permissions and limitations under the License.
...
 Alexis Conneau.
 
-An example of how to use a [`SpeechEncoderDecoderModel`] for inference can be seen in
-[Speech2Text2](speech_to_text_2).
+An example of how to use a [`SpeechEncoderDecoderModel`] for inference can be seen in [Speech2Text2](speech_to_text_2).
 
+## Randomly initializing `SpeechEncoderDecoderModel` from model configurations.
+
+[`SpeechEncoderDecoderModel`] can be randomly initialized from an encoder and a decoder config. In the following example, we show how to do this using the default [`Wav2Vec2Model`] configuration for the encoder
+and the default ['BertModel`] configuration for the decoder.
```
Comment (patrickvonplaten) [inline docs/source/en/model_doc/speech-encoder-decoder.mdx:28]:
```suggestion
and the default [`BertForCausalLM`] configuration for the decoder.
```

File: docs/source/en/model_doc/speech-encoder-decoder.mdx [hunk 5]
```diff
@@ -12,17 +12,103 @@ specific language governing permissions and limitations under the License.
...
+Depending on which architecture you choose as the decoder, the cross-attention layers might be randomly initialized.
+Initializing [`SpeechEncoderDecoderModel`] from a pretrained encoder and decoder checkpoint requires the model to be fine-tuned on a downstream task, as has been shown in [the *Warm-starting-encoder-decoder blog post*](https://huggingface.co/blog/warm-starting-encoder-decoder).
+To do so, the `SpeechEncoderDecoderModel` class provides a [`SpeechEncoderDecoderModel.from_encoder_decoder_pretrained`] method.
+
+```python
+>>> from transformers import SpeechEncoderDecoderModel
+
+>>> model = SpeechEncoderDecoderModel.from_encoder_decoder_pretrained(
+...     "facebook/hubert-large-ls960-ft", "bert-base-uncased"
```
Comment (patrickvonplaten) [inline docs/source/en/model_doc/speech-encoder-decoder.mdx:51]:
```suggestion
...     "facebook/hubert-large-ll60k", "bert-base-uncased"
```

Comment (patrickvonplaten) [reply docs/source/en/model_doc/speech-encoder-decoder.mdx:51]: Let's not use a fine-tuned checkpoint here but just a pretrained one

**Review by patrickvonplaten — APPROVED** [review]
Only checked the speech part - looks very nice!

**Pull request merged**